warehouse/forklift/legacy.py

```diff
@@ -649,13 +690,23 @@ def file_upload(request):
         else:
             field_name = sorted(form.errors.keys())[0]
 
-        raise _exc_with_message(
-            HTTPBadRequest,
-            "{field}: {msgs[0]}".format(
-                field=field_name,
-                msgs=form.errors[field_name],
-            ),
-        )
+        if field_name in form:
+            raise _exc_with_message(
+                HTTPBadRequest,
+                "{value!r} is an invalid value for {field}.\n".format(
+                    value=form[field_name].data,
+                    field=form[field_name].label.text) +
+                "Error: {}\n".format(form.errors[field_name][0]) +
+                "see "
+                "https://packaging.python.org/specifications/core-metadata/",
+            )
```

Comment:
Could you remove the newlines from this error message?

Reply:
Removing the newline made the line longer than 80 characters, should I break it again?

Reply:
By newlines, I mean the "\n" characters you have in this string.

Reply:
Totally my bad, sorry! I just removed the `\n`s 😅